src/core/thread/mle_router.cpp

```diff
@@ -3963,6 +3966,8 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Header *aHeader, Message *aMe
     Router *router;
     bool old;
 
+    mAddressSolicitPending = false;
```

Comment:
also reset `mAddressSolicitPending` after some time? There might be chances that the address solicit response is lost.

Reply:
The response handler should also be called when the CoAP response timeout is reached.  Are you thinking of having a separate timeout within MLE just to be safe?

Reply:
I didn't think of the CoAP response timeout. It should be fine to keep it as it is. Thanks.